var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -62,50 +63,73 @@ class Mfem(Package):
+    variant('suite-sparse', default=False,
+        description='Enable serial, sparse direct solvers')
+    variant('petsc', default=False,
+        description='Enable PETSc solvers, preconditioners, etc..')
+    variant('sundials', default=False,
+        description='Enable Sundials time integrators')
+    variant('mpfr', default=False,
+        description='Enable precise, 1D quadrature rules')
+    variant('lapack', default=False,
+        description='Use external blas/lapack routines')
+    variant('debug', default=False,
+        description='Build debug instead of optimized version')
+    variant('netcdf', default=False,
+        description='Enable Cubit/Genesis reader')
+    variant('gzstream', default=True,
+        description='Support zip\'d streams for I/O')
+    variant('examples', default=False,
+        description='Build and install examples')
+    variant('minapps', default=False,
+        description='Build and install minapps')
 
     depends_on('blas', when='+lapack')
+    depends_on('blas', when='+suite-sparse')
     depends_on('lapack', when='+lapack')
+    depends_on('lapack', when='+suite-sparse')
 
     depends_on('mpi', when='+mpi')
-    depends_on('metis', when='+mpi')
-    depends_on('hypre', when='+mpi')
-
-    depends_on('hypre', when='+hypre')
-
-    depends_on('metis@4:', when='+metis')
-
-    depends_on('suite-sparse', when='+suite-sparse')
-    depends_on('blas', when='+suite-sparse')
-    depends_on('lapack', when='+suite-sparse')
+    depends_on('metis')
+    depends_on('parmetis', when='+superlu-dist')
+    depends_on('metis@5:', when='+superlu-dist')
     depends_on('metis@5:', when='+suite-sparse ^suite-sparse@4.5:')
-    depends_on('cmake', when='^metis@5:', type='build')
+    depends_on('hypre~internal-superlu', when='+mpi')
+    depends_on('hypre@develop~internal-superlu', when='+petsc +hypre')
 
+    depends_on('sundials@2.7:+hypre', when='+sundials')
+    depends_on('suite-sparse', when='+suite-sparse')
     depends_on('superlu-dist', when='@3.2: +superlu-dist')
+    depends_on('petsc@develop', when='+petsc')
 
+    depends_on('mpfr', when='+mpfr')
+    depends_on('cmake', when='^metis@5:', type='build')
     depends_on('netcdf', when='@3.2: +netcdf')
     depends_on('zlib', when='@3.2: +netcdf')
     depends_on('hdf5', when='@3.2: +netcdf')
+    depends_on('libunwind', when='+debug')
+    depends_on('zlib', when='+gzstream')
 
     patch('mfem_ppc_build.patch', when='@3.2:3.3 arch=ppc64le')
 
     def check_variants(self, spec):
-        if '+mpi' in spec and ('+hypre' not in spec or '+metis' not in spec):
-            raise InstallError('mfem+mpi must be built with +hypre ' +
-                               'and +metis!')
-        if '+suite-sparse' in spec and ('+metis' not in spec or
-                                        '+lapack' not in spec):
+        if '+mpi' in spec and '+hypre' not in spec:
+            raise InstallError('mfem+mpi must be built with +hypre')
```

Comment:
Hmm...I am on the fence about that. I mean, these lines pre-date any work I've done on this package. If they are/were wrong then, why were they there to begin with? Its probably the case that `conflicts()` functionality was added after these lines were introduced. Is that right?

So, when I go into Spack and add feature `xyz` to a package, does that obligate me to adjust other parts of the package's code that I didn't originally write? Does it obligate me to update the package code to newer syntax/functionality that was added to Spack since the package was last updated?

Here, I guess I've made enough changes that taking ownership of this portion of the package is probably the right thing to do. Having said that, I didn't touch these lines except where it was necessary to get additional variants I added working specifically because they were a) already approved in a previous PR and b) I wasn't aware that there is a new way to do this.